src/ui/components/search/filtersearchcomponent.js

```diff
@@ -128,7 +115,7 @@ export default class FilterSearchComponent extends Component {
   // TODO(oshi): SPR-1925 check that it is safe to remove this, it runs an extra search
   // For no obvious reasons
   onCreate () {
-    if (this.query && this.filter) {
+    if (this.query) {
```

Comment:
For backwards compatibility, I think we need to add a new config option called "executeInitialSearch" or something like that. That option will have the following behavior:

this.executeInitialSearch = config.executeInitialSearch || config.filter

Not sure if "executeInitialSearch" is the best name. If we just check for this.query here, it could alter the behavior of existing sites. A client may have a config.query, but not config.filter, and is not expecting a network call to be issued when this component is created. We should update the README too.

Reply:
Sure that sgtm!

Reply:
added it as searchOnLoad